DVR/URLRequest.swift

```diff
@@ -25,6 +25,15 @@ extension NSURLRequest {
 }
 
 
+extension NSURLRequest {
+    func requestByAppendingHeaders(headers: [NSObject: AnyObject]) -> NSURLRequest {
+        let request = mutableCopy() as! NSMutableURLRequest
```

Comment:
Ya that's exactly right. I did this here for simplicity and because if this fails, we have way bigger problems.